Sources/ClientGtk/ServerListView.swift

```diff
@@ -0,0 +1,93 @@
+import SwiftCrossUI
+import DeltaCore
+
+enum DetailState {
+  case addition, server(ServerDescriptor), error(String)
+}
+
+class ServerListViewState: Observable {
+  @Observed var servers: [ServerDescriptor] = []
+  @Observed var detailState = DetailState.addition
+}
+
+struct ServerListView: View {
+  var completionHandler: (ServerDescriptor) -> Void
+
+  var state = ServerListViewState()
+
+  var body: some ViewContent {
+    NavigationSplitView {
+      VStack {
+        Button("Add server") {
+          state.detailState = .addition
+        }
+
+        ScrollView {
+          ForEach(state.servers) { server in
+            Button(server.name) {
+              state.detailState = .server(server)
+            }
+            .padding(Edge.Set.bottom, 5)
+          }
+        }
+      }
+      .frame(minWidth: 100)
+      .padding(Edge.Set.trailing, 10)
+    } detail: {
+      VStack {
+        switch state.detailState {
+          case .addition:
+            additionView
+          case .server(let server):
+            Text(server.name)
+            if let port = server.port {
+              Text(server.host + ":" + String(port))
+            } else {
+              Text(server.host)
+            }
+
+            Button("Connect") {
+              completionHandler(server)
+            }
+          case .error(let message):
+            Text(message)
+        }
+      }
+      .padding(Edge.Set.leading, 10)
+    }
+  }
+
+  var additionView: some View {
+    VStack {
+      @Observed var name = ""
+      @Observed var address = ""
+
+      Text("Add server")
+      TextField("Name", $name)
+      TextField("Address", $address)
+      
+      Button("Add") {
+        let parts = address.split(separator: ":")
+        guard parts.count <= 2 else {
```

Comment:
Change to `parts.count > 0, parts.count <= 2` to account for the user inputting an empty address. Although the crash is currently preventing that case from being reached, it's good practice to still account for it as the code would be more likely to remain correct even after being refactored or after the crash in stdlib is fixed.

Reply:
Sure, again, I didn’t change this from the previous implementation, will do.

Reply:
Ah, yeah sorry didn't realise that this was from the previous file